packages/db/schema.prisma

```diff
@@ -419,3 +431,25 @@ model BetaTokens {
 
   @@index([userId])
 }
+
+model Notification {
+  id               Int              @id @default(autoincrement())
+  createdAt        DateTime         @default(now())
+  updatedAt        DateTime         @updatedAt
+  type             NotificationType
+  toUser           User             @relation("notificationToUser", fields: [toUserId], references: [id])
+  toUserId         String
+  fromUser         User             @relation("notificationFromUser", fields: [fromUserId], references: [id])
+  fromUserId       String
+  sharedSolutionId Int?
```

Comment:
Have we considered deduplication or aggregation of notifications? This would be particularly useful in scenarios where multiple similar notifications occur within a short span of time. For example, when multiple users upvote a solution, like a comment, or reply to a comment, instead of creating a separate entry in the notification table for each action (which could result in an overwhelming number of notifications), we could aggregate or group similar notifications.

Reply:
No need ATM, we don't have Taylor Swift posting on our platform 😭

Reply:
probably something to consider post MVP of this

Reply:
i would like to know how we can accomplish that though @kfahad5607